nixos/modules/services/continuous-integration/gocd-server/default.nix

```diff
@@ -0,0 +1,163 @@
+        default = [ pkgs.stdenv pkgs.jre7 config.programs.ssh.package pkgs.nix ];
+        type = types.listOf types.package;
+        description = ''
+          Packages to add to PATH for the gocd-server process.
+        '';
+      };
+
+      debug = mkOption {
+        default = "";
+        example = "-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=5005";
+        type = types.str;
+        description = ''
+          Specifies the JVM debug options.  By default debug is disabled.
+        '';
+      };
+
+      environment = mkOption {
+        default = { };
+        type = with types; attrsOf str;
+        description = ''
+          Additional environment variables to be passed to the gocd-server process.
+          As a base environment, gocd-server receives NIX_PATH from
+          <option>environment.sessionVariables</option>, NIX_REMOTE is set to
+          "daemon".
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.extraGroups = optional (cfg.group == "gocd-server") {
+      name = "gocd-server";
+      gid = config.ids.gids.gocd-server;
+    };
+
+    users.extraUsers = optional (cfg.user == "gocd-server") {
+      name = "gocd-server";
+      description = "gocd-server user";
+      createHome = true;
+      home = cfg.workDir;
+      group = cfg.group;
+      extraGroups = cfg.extraGroups;
+      useDefaultShell = true;
+      uid = config.ids.gids.gocd-server;
+    };
+
+    systemd.services.gocd-server = {
+      description = "GoCD Server";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      environment =
+        let
+          selectedSessionVars =
+            lib.filterAttrs (n: v: builtins.elem n [ "NIX_PATH" ])
+              config.environment.sessionVariables;
+        in
+          selectedSessionVars //
+            { NIX_REMOTE = "daemon";
+            } //
+            cfg.environment;
+
+      path = cfg.packages;
+
+      script = ''
+        ${pkgs.jre7}/bin/java -server \
+                              -Xms512m \
+                              -Xmx1024m \
+                              -XX:PermSize=128m \
+                              -XX:MaxPermSize=256m \
+                              ${cfg.debug} \
+                              -verbose:gc \
+                              -Xloggc:go-server-gc.log \
+                              -XX:+PrintGCTimeStamps \
+                              -XX:+PrintTenuringDistribution \
+                              -XX:+PrintGCDetails \
+                              -XX:+PrintGC \
+                              -Dcruise.listen.host=${cfg.listenAddress} \
+                              -Duser.language=en \
+                              -Djruby.rack.request.size.threshold.bytes=30000000 \
+                              -Duser.country=US \
+                              -Dcruise.config.dir=/etc/go \
+                              -Dcruise.config.file=/etc/go/cruise-config.xml \
```

Comment:
'cruise.config.dir' is the location that the gocd server stores all of it's application configuration files.  I will create a 'conf' directory inside of the package root directory and point 'cruise.config.dir' and 'cruise.config.file' to that 'conf' directory.